sotodlib/io/load_smurf.py

```diff
@@ -1005,10 +1022,189 @@ def update_observation_files(self, obs, session, max_early=5, max_wait=100, forc
+                continue
+
+            for stream_id in sorted(os.listdir( os.path.join(self.meta_path, ct_dir)), reverse=reverse):
+                    action_path = os.path.join(self.meta_path, ct_dir, stream_id)
+                    actions = sorted(os.listdir( action_path ), reverse=reverse)
+
+                    for action in actions:
+                        try:
+                            ctime = int( action.split('_')[0] )
+                            if ctime < min_ctime or ctime > max_ctime:
+                                continue
+                            astring = '_'.join(action.split('_')[1:])
+
+                            yield (astring, stream_id, ctime, os.path.join(action_path, action))
+                        except GeneratorExit:
+                            return
+                        except:
+                            continue
+
+    def search_metadata_files(self,  min_ctime=16000*1e5, max_ctime=None, reverse=False,
+                          skip_plots=True, skip_configs=True):
+        """Generator used to page through smurf folder returning each file formatted for
+            easy use. 
+
+        Args
+        -----
+        min_ctime : int or float
+            Lowest timestamped action to return
+        max_ctime : int or float
+            highest timestamped action to return
+        reverse   : bool
+            if true, goes backward 
+        skip_plots : bool
+            if true, skips all the plots folders because we probably don't want to look through them
+        skip_configs : bool
+            if true, skips all the config folders because we probably don't want to look through them
+
+        Yields
+        -------
+        tuple (fname, ctime, abs_path)
+        fname : string
+            file name with ctime removed
+        ctime : int
+            file ctime
+        abs_path : string
+            absolute path to file
+        """
+        for action, stream_id, actime, path in self.search_metadata_actions(min_ctime=min_ctime,
+                                                                            max_ctime=max_ctime,
+                                                                            reverse=reverse):
+            if skip_configs and action == 'config':
+                continue
+            adirs = os.listdir(path)
+            for adir in adirs:
+                if skip_plots and adir == 'plots':
+                    continue
+                for root, dirs, files in os.walk(os.path.join(path, adir), topdown=False):
+                    for name in files:
+                        try:
+                            try:
+                                ctime = int(name.split('_')[0])
+                            except ValueError:
+                                ctime = actime
+                            fname = '_'.join(name.split('_')[1:])
+                            yield (fname, stream_id, ctime, os.path.join(root, name))
```

Comment:
This method of returning ctimes + filenames seems strange to me, and doesn't really work if the first component of the filename is not a ctime (i.e. you will still be throwing away the first word). Maybe this should return the ctime and also the full filename including the first word?

Reply:
I think the big thing here is the fname without the ctime makes for fast/easier pattern matching scripts. Matches similarly with the searching through the folders. Also, all the standard files from smurf all come with the ctime in front. I'd expect that to remain a requirement.